### PR TITLE
split elf::Sym definition out of alloc gate

### PR DESCRIPTION
**Pull request by lzutao**
Also derive Copy trait because sym{32,64}::Sym have it
See the non-whitespace diff: https://github.com/m4b/goblin/pull/198/files?utf8=%E2%9C%93&diff=unified&w=1

### CODE REVIEW

File: src/elf/sym.rs [hunk 1]
```diff
@@ -320,112 +318,111 @@ pub mod sym64 {
...
+                sym32::SIZEOF_SYM
+            },
+            Container::Big => {
+                sym64::SIZEOF_SYM
+            },
         }
     }
+}
 
+if_alloc! {
```
Comment (m4b) [inline src/elf/sym.rs]: How come the TryFromCtx impls require alloc ?

Comment (lzutao) [reply src/elf/sym.rs]: The error::Error has variant `Malformed(String)` which requires allocation.

Comment (m4b) [reply src/elf/sym.rs]: Ah yea. Hmmm, might be worth investigating whether static strs could replace or more variants. Anyway thanks for PR!

**Comment by philipc** [comment]
This is fine if it is useful to you, but it seems pointless without `impl From<Sym> for ElfSym` and `impl From<ElfSym> for Sym`. Can you give your motivation for why this is useful, and why only for `Sym` (there's other unified structs that are currently alloc only too)?

**Comment by lzutao** [comment]
I am trying to unlock GnuHash in #151 , the GnuHash::find return a Sym
but GnuHash itself is not under alloc gate. So the test failed.

> it seems pointless without `impl From<Sym> for ElfSym` and `impl From<ElfSym> for Sym`

I would move that out of alloc gate too.

File: src/elf/sym.rs [hunk 2]
```diff
@@ -163,93 +163,92 @@ macro_rules! elf_sym_std_impl {
...
-            pub unsafe fn from_raw<'a>(symp: *const Sym, count: usize) -> &'a [Sym] {
-                slice::from_raw_parts(symp, count)
-            }
+        #[inline]
+        pub unsafe fn from_raw<'a>(symp: *const Sym, count: usize) -> &'a [Sym] {
+            slice::from_raw_parts(symp, count)
+        }
+
+        if_alloc! {
+            if_std! {
```
Comment (philipc) [inline src/elf/sym.rs]: I think you can delete the `if_alloc! {}` if everything inside it is gated by `if_std!`.

Comment (lzutao) [reply src/elf/sym.rs]: Yeah, that's right

**Pull request merged**